### PR TITLE
ENYO-871: Imported resolution independence code from moonstone and added...

### PR DESCRIPTION
**Pull request by Djspaceg**
... some initialization and externalization features so any visual library can establish various screen resolutions their interface is targeting.

* Added `enyo.ri.defineScreenTypes`.
* Privatized the `screenTypes` array, and standardized the internal var names.

Enyo-DCO-1.1-Signed-off-by: Blake Stephens <blake.stephens@lge.com>

### CODE REVIEW

File: source/ui/resolution.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		* This will need to be re-run any time the screen size changes, so all the values can be
+		* re-cached.
+		*
+		* @public
+		*/
+		// Later we can wire this up to a screen resize event so it doesn't need to be called manually.
+		init: function () {
+			_screenType = this.getScreenType();
+			this.updateScreenTypeOnBody();
+			enyo.dom.unitToPixelFactors.rem = scope.enyo.getUnitToPixelFactors();
```
Comment (aarontam) [inline source/ui/resolution.js]: Should this be `enyo.ri.getUnitToPixelFactors()`?

File: source/ui/resolution.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+			return 1;
+		},
+
+		/**
+		* @private
+		*/
+		getUnitToPixelFactors: function (type) {
+			type = type || _screenType;
+			if (type) {
+				_riRatio = getScreenTypeObject(type).pxPerRem;
```
Comment (aarontam) [inline source/ui/resolution.js]: Is this line intentional - seems to be missing a divisor, potentially. If I call `enyo.ri.scale(1)` more than once, the result of the first call is `1` and subsequent calls are `24`, in FHD.

File: source/ui/resolution.js (hunk 1 above)
Comment (Djspaceg) [inline source/ui/resolution.js]: I caught this too late and it was corrected in 958. It should have been fixed in this branch though.

Comment (aarontam) [inline source/ui/resolution.js]: Ah ha! Maybe I should have started with reviewing 958 instead - was trying to do them discretely to make the review more tractable. :smiley:

File: source/ui/resolution.js (hunk 2 above)
Comment (Djspaceg) [inline source/ui/resolution.js]: This does seem to be a general problem that didn't reveal itself earlier. I'll look into it.

**Pull request merged**